docs/reference/mapping/params/eager-global-ordinals.asciidoc

```diff
@@ -0,0 +1,76 @@
+[[eager-global-ordinals]]
+=== `eager_global_ordinals`
+
+Global ordinals is a data-structure on top of doc values, that maintains an
+incremental numbering for each unique term in a lexicographic order. Each
+term has a unique number and the number of term 'A' is lower than the
+number of term 'B'. Global ordinals are only supported on
+<<keyword,`keyword`>> fields, as well as <<text,`text`>> fields when
+<<fielddata,`fielddata`>> is enabled.
+
+Doc values (and fielddata) also have ordinals, which is a unique numbering for
+all terms in a particular segment and field. Global ordinals just build on top
+of this, by providing a mapping between the segment ordinals and the global
+ordinals, the latter being unique across the entire shard.
+
+Global ordinals are used for features that use segment ordinals, such as
+the <<search-aggregations-bucket-terms-aggregation,`terms` aggregation>>,
+to improve the execution time. A terms aggregation relies purely on global
+ordinals to perform the aggregation at the shard level, then converts global
+ordinals to the real term only for the final reduce phase, which combines
+results from different shards.
+
+Global ordinals for a specified field are tied to _all the segments of a
```

Comment:
I feel like this can be combined with the paragraph above starting with "doc values (and fielddata)". I feel like both are about how you get the global ordinals the the paragraph between the two is about what they are good for.